src/qtum/qtumx86.cpp

```diff
@@ -371,6 +372,22 @@ uint32_t QtumHypervisor::ReadExternalStorage(uint32_t syscall, x86Lib::x86CPU& v
     return status;
 }
 
+uint32_t QtumHypervisor::SHA256(uint32_t syscall, x86Lib::x86CPU& vm) {
+    //eax = success
+    //ebx = original value
+    //ecx = sizeof original value
+    //edx = hash of original value
+    size_t len = vm.Reg32(ECX);
+    unsigned char* k = new unsigned char[len];
+    unsigned char hash[CSHA256::OUTPUT_SIZE] = {};
+    vm.ReadMemory(vm.Reg32(EBX), len, k);
+    CSHA256().Reset().Write(k, len-1).Finalize(hash); // create hash
```

Comment:
Why `len-1` ?

Reply:
avoids a buffer overrun methinks. I basically tried to mimick the inputs that I was getting when I originally had it working and tried to cut out all the casting and different types and crap. I could try and see if it affects it.